domain/status/state/state.go

```diff
@@ -40,6 +41,40 @@ func NewState(factory database.TxnRunnerFactory, clock clock.Clock, logger logge
 	}
 }
 
+// GetModelInfo returns the model's basic information.
+func (st *State) GetModelInfo(ctx context.Context) (status.ModelStatusInfo, error) {
+	type modelInfo struct {
+		ModelType       string `db:"type"`
+		CredentialOwner string `db:"credential_owner"`
+	}
+
+	db, err := st.DB()
+	if err != nil {
+		return status.ModelStatusInfo{}, errors.Capture(err)
+	}
+
+	stmt, err := st.Prepare(`
+SELECT &modelInfo.*
+FROM   model
+`, modelInfo{})
+	if err != nil {
+		return status.ModelStatusInfo{}, errors.Capture(err)
+	}
+
+	var m modelInfo
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		return tx.Query(ctx, stmt).Get(&m)
```

Comment:
The rest of this state file establishes the pattern of checking for a no rows error and returning a model not found error if this occurs. This gives us signal about the state of the database when the call was made. I think we should do the same here.

Reply:
Resolved offline